### PR TITLE
Update Vundle Plugin to latest oh-my-fish package specs and use the latest Vundle.vim

### PR DESCRIPTION
**Pull request by nwykes**

### CODE REVIEW

**Comment by derekstavis** [comment]
Thanks for the patch! As @bpinto also commented in https://github.com/oh-my-fish/oh-my-fish/pull/28, this plugin makes the plugin install Vundle for you if not installed, and we prefer keeping this to the user. Would you mind replacing the offending lines by a check and possibly a message to the user?

**Comment by nwykes** [comment]
Can do.

File: vundle.fish [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+function vundle-update
+  vim -c "execute \"PluginUpdate\" | q | q"
+end
+
+function vundle-clean
+  vim -c "execute \"PluginClean!\" | q | q"
+end
+
+function uninstall --on-event uninstall_vundle
```
Comment (bpinto) [inline vundle.fish]: This function does nothing, it can be removed.

**Comment by bpinto** [comment]
Too many functions are defined inside vundle.fish, we should split them by creating one file for each function.

**Comment by nwykes** [comment]
I believe I've addressed the comments.  @bpinto and @derekstavis please take a look.  Let me know if you'd like the commits squashed.

**Comment by derekstavis** [comment]
Squashing the commits and rebasing in current master (if needed) would be greater. At the rest the PR looks good to me.

**Comment by nwykes** [comment]
Thanks @derekstavis.  I think I misread your comment at first and squashed my fork, but have reset it so you can do with it as you will.

**Comment by derekstavis** [comment]
You can squash them into a single commit named "Update to new OMF pkg format" and push to the same branch to update this PR.

**Comment by derekstavis** [comment]
Thanks @nwykes. LGTM! :+1: what about now @bpinto?

File: vundle.fish [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# SYNOPSIS
+#   Upon installation or loading, vundle checks for the existence
+#   of the Vundle.vim plugin and will notify if it is missing.
+#
+# USAGE
+#   $ vundle # runs PluginInstall to install missing plugins
```
Comment (bpinto) [inline vundle.fish:6]: I have never seen an usage with a description, it seems like the explanation is part of the synopsis and not part of the usage.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+```fish
+$ omf install vundle
+```
+
+## Usage
+
+Upon installation or loading, vundle checks for the existence of the Vundle.vim plugin and will notify if it is missing.
+
+```fish
+$ vundle # runs PluginInstall to install missing plugins
```
Comment (bpinto) [inline README.md:24]: We could have used the oh-my-fish's readme usage section layout.

**Comment by bpinto** [comment]
Thanks. I have added two comments that would be nice to come in a future PR but are not show stoppers.

**Pull request merged**